File: native/strapiclient/src/model/adopted_cat.rs
use crate::model::Cat;
use rustler::NifStruct;

use crate::primitives::DateTime;

#[derive(Debug, NifStruct)]
#[module = "Kotkowo.AdoptedCat"]
pub struct AdoptedCat {
    pub id: usize,
    pub cat: Cat,
    pub adoption_date: DateTime,
}

File: lib/kotkowo/client/adopted_cat.ex
defmodule Kotkowo.Client.AdoptedCat do
  @moduledoc false
  alias Kotkowo.Client.Cat

  defstruct [
    :id,
    :adoption_date,
    :cat
  ]

  @type t() :: %__MODULE__{
          id: String.t() | nil,
          cat: Cat.t(),
          adoption_date: Kotkowo.Client.DateTime.t()
        }
end
